core/src/main/kotlin/org/evomaster/core/mongo/filter/DocumentToASTFilterConverter.kt

```diff
@@ -0,0 +1,747 @@
+package org.evomaster.core.mongo.filter
+
+import org.bson.BsonRegularExpression
+import org.bson.BsonType
+import org.bson.Document
+
+class DocumentToASTFilterConverter {
```

Comment:
which version of MongoDB are we targeting here? 
I know that there have been some breaking changes at some point.
We can just support a single version (perhaps the most recent?).
But somewhere we need to have a variable specifying the supported version (in case we want to support different versions in the future).

Reply:
we could start supporting 3.2, and if we need, also support 4.2.2